sympy/core/numbers.py

```diff
@@ -1294,73 +1321,60 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self == other
 
-    def __gt__(self, other):
+    def _Frel(self, other, op):
         try:
             other = _sympify(other)
         except SympifyError:
             raise TypeError("Invalid comparison %s > %s" % (self, other))
         if other.is_NumberSymbol:
-            return other.__lt__(self)
-        if other.is_Rational and not other.is_Integer:
-            self *= other.q
-            other = _sympify(other.p)
-        elif other.is_comparable:
-            other = other.evalf()
-        if other.is_Number and other is not S.NaN:
+            return other
+        elif other.is_Rational:
+            # if you don't use self._prec here, the Rational
+            # will be choppped to default precision
+            other = Float(other, self._prec)
```

Comment:
Would it be possible to use the same method as above:

        if other.is_Rational and not other.is_Integer:
            self *= other.q
            other = _sympify(other.p)

Then the full precision of `other` would be preserved.

Reply:
This is not for equality here so I am just scaling them to the same precision. There was also a test involving a gargantuan number (`E**1000`) that is making me defensive in not wanting to increase in size when dealing with Floats.